<a href="https://colab.research.google.com/github/Sana-0511/Document_Summarization-/blob/main/BART.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# BART

In [ ]:
pip install transformers

In [ ]:
# Install the required utility
!apt-get install wget

# Import the necessary libraries
import zipfile
import os

# Provide the Google Drive link
google_drive_link = 'https://drive.google.com/uc?id=1FV5aYhpvl6NFrH7aYzEVg8pmO_yokYoI'

# Specify the output file name
zip_file = 'example.zip'

# Download the zip file
!wget --no-check-certificate $google_drive_link -O $zip_file

# Create a directory to extract the contents
extracted_dir = 'extracted_content'
os.makedirs(extracted_dir, exist_ok=True)

# Extract the contents of the zip file
with zipfile.ZipFile(zip_file, 'r') as zip_ref:
    zip_ref.extractall(extracted_dir)

# List the files in the extracted directory
extracted_files = os.listdir(extracted_dir)
print("Files extracted:", extracted_files)

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
wget is already the newest version (1.21.2-2ubuntu1).
0 upgraded, 0 newly installed, 0 to remove and 30 not upgraded.
--2024-01-23 05:04:03--  https://drive.google.com/uc?id=1FV5aYhpvl6NFrH7aYzEVg8pmO_yokYoI
Resolving drive.google.com (drive.google.com)... 173.194.212.100, 173.194.212.138, 173.194.212.113, ...
Connecting to drive.google.com (drive.google.com)|173.194.212.100|:443... connected.
HTTP request sent, awaiting response... 303 See Other
Location: https://drive.usercontent.google.com/download?id=1FV5aYhpvl6NFrH7aYzEVg8pmO_yokYoI [following]
--2024-01-23 05:04:03--  https://drive.usercontent.google.com/download?id=1FV5aYhpvl6NFrH7aYzEVg8pmO_yokYoI
Resolving drive.usercontent.google.com (drive.usercontent.google.com)... 74.125.196.132, 2607:f8b0:400c:c36::84
Connecting to drive.usercontent.google.com (drive.usercontent.google.com)|74.125.196.132|:443... connected.
HTTP request sent, 

In [ ]:
import torch
from transformers import BartForConditionalGeneration, BartTokenizer

# Load pre-trained BART tokenizer and model
tokenizer = BartTokenizer.from_pretrained('facebook/bart-large-cnn')
model = BartForConditionalGeneration.from_pretrained('facebook/bart-large-cnn')

## Generate Method

In [ ]:
# Load the content of the text file
file_path = '/content/extracted_content/example/msft.txt'
with open(file_path, 'r', encoding='utf-8') as file:
    text = file.read()

# Tokenize and encode the input text
inputs = tokenizer.encode("summarize: " + text, return_tensors="pt", max_length=1024, truncation=True)

# Generate the summary using the BART model
summary_ids = model.generate(inputs, max_length=150, length_penalty=2.0, num_beams=4, early_stopping=True)

# Decode the generated summary and print it
summary_text = tokenizer.decode(summary_ids[0], skip_special_tokens=True)
print("Original Text:")
print(text)
print("\nSummary:")
print(summary_text)

Original Text:
In an attempt to build an AI-ready workforce, Microsoft announced Intelligent Cloud Hub which has been launched to empower the next generation of students with AI-ready skills. Envisioned as a three-year collaborative program, Intelligent Cloud Hub will support around 100 institutions with AI infrastructure, course content and curriculum, developer support, development tools and give students access to cloud and AI services. As part of the program, the Redmond giant which wants to expand its reach and is planning to build a strong developer ecosystem in India with the program will set up the core AI infrastructure and IoT Hub for the selected campuses. The company will provide AI development tools and Azure AI services such as Microsoft Cognitive Services, Bot Services and Azure Machine Learning.According to Manish Prakash, Country General Manager-PS, Health and Education, Microsoft India, said, "With AI being the defining technology of our time, it is transforming lives

Original Text:
In an attempt to build an AI-ready workforce, Microsoft announced Intelligent Cloud Hub which has been launched to empower the next generation of students with AI-ready skills. Envisioned as a three-year collaborative program, Intelligent Cloud Hub will support around 100 institutions with AI infrastructure, course content and curriculum, developer support, development tools and give students access to cloud and AI services. As part of the program, the Redmond giant which wants to expand its reach and is planning to build a strong developer ecosystem in India with the program will set up the core AI infrastructure and IoT Hub for the selected campuses. The company will provide AI development tools and Azure AI services such as Microsoft Cognitive Services, Bot Services and Azure Machine Learning.According to Manish Prakash, Country General Manager-PS, Health and Education, Microsoft India, said, "With AI being the defining technology of our time, it is transforming lives and industry and the jobs of tomorrow will require a different skillset. This will require more collaborations and training and working with AI. That’s why it has become more critical than ever for educational institutions to integrate new cloud and AI technologies. The program is an attempt to ramp up the institutional set-up and build capabilities among the educators to educate the workforce of tomorrow." The program aims to build up the cognitive skills and in-depth understanding of developing intelligent cloud connected solutions for applications across industry. Earlier in April this year, the company announced Microsoft Professional Program In AI as a learning track open to the public. The program was developed to provide job ready skills to programmers who wanted to hone their skills in AI and data science with a series of online courses which featured hands-on labs and expert instructors as well. This program also included developer-focused AI school that provided a bunch of assets to help build AI skills.

Summary:
Intelligent Cloud Hub will support around 100 institutions with AI infrastructure, course content and curriculum. Redmond giant wants to expand its reach and is planning to build a strong developer ecosystem in India. The program aims to build up the cognitive skills and in-depth understanding of developing intelligent cloud connected solutions for applications across industry.

ROUGE scores provide information on precision (p), recall (r), and F1-score (f) for ROUGE-1, ROUGE-2, and ROUGE-L metrics.

These scores provide an evaluation of the generated summary against the original text. In general:

- Precision measures how many of the generated n-grams were correct.
- Recall measures how many of the original n-grams were captured by the generated summary.
- F1-score is the harmonic mean of precision and recall and provides a balanced measure.

In [ ]:
!pip install rouge
from rouge import Rouge

# Your original text and generated summary
original_text = text
generated_summary = summary_text

# Initialize the Rouge object
rouge = Rouge()

# Calculate ROUGE scores
scores = rouge.get_scores(generated_summary, original_text)

# Print the scores
print("ROUGE Scores:")
print(scores)

ROUGE Scores:
[{'rouge-1': {'r': 0.27071823204419887, 'p': 0.98, 'f': 0.42424242085043384}, 'rouge-2': {'r': 0.16666666666666666, 'p': 0.9245283018867925, 'f': 0.2824207466913603}, 'rouge-l': {'r': 0.27071823204419887, 'p': 0.98, 'f': 0.42424242085043384}}]


ROUGE-1:

Precision (p): 0.980
Recall (r): 0.271
F1-score (f): 0.424

ROUGE-2:

Precision (p): 0.925
Recall (r): 0.167
F1-score (f): 0.282

ROUGE-L:

Precision (p): 0.980
Recall (r): 0.271
F1-score (f): 0.424

These scores indicate that the generated summary has high precision but lower recall. In other words, the generated summary includes a significant portion of correct n-grams, but it might miss some important content present in the original text.

In [ ]:
# Load the content of the text file
file_path = '/content/extracted_content/example/computer.txt'
with open(file_path, 'r', encoding='utf-8') as file:
    text = file.read()

# Tokenize and encode the input text
inputs = tokenizer.encode("summarize: " + text, return_tensors="pt", max_length=1024, truncation=True)

# Generate the summary using the BART model
summary_ids = model.generate(inputs, max_length=150, length_penalty=2.0, num_beams=4, early_stopping=True)

# Decode the generated summary and print it
summary_text = tokenizer.decode(summary_ids[0], skip_special_tokens=True)
print("Original Text:")
print(text)
print("\nSummary:")
print(summary_text)

Original Text:
A computer is a machine that can be programmed to carry out sequences of arithmetic or logical operations automatically. Modern digital electronic computers can perform generic sets of operations known as programs. These programs enable computers to perform a wide range of tasks. A computer system is a nominally complete computer that includes the hardware, operating system, and peripheral equipment needed and used for full operation. This term may also refer to a group of computers that are linked and function together, such as a computer network or computer cluster. A broad range of industrial and consumer products use computers as control systems. Simple special-purpose devices like microwave ovens and remote controls are included, as are factory devices like industrial robots and computer-aided design, as well as general-purpose devices like personal computers and mobile devices like smartphones. Computers power the Internet, which links billions of other computers a

Original Text:
A computer is a machine that can be programmed to carry out sequences of arithmetic or logical operations automatically. Modern digital electronic computers can perform generic sets of operations known as programs. These programs enable computers to perform a wide range of tasks. A computer system is a nominally complete computer that includes the hardware, operating system, and peripheral equipment needed and used for full operation. This term may also refer to a group of computers that are linked and function together, such as a computer network or computer cluster. A broad range of industrial and consumer products use computers as control systems. Simple special-purpose devices like microwave ovens and remote controls are included, as are factory devices like industrial robots and computer-aided design, as well as general-purpose devices like personal computers and mobile devices like smartphones. Computers power the Internet, which links billions of other computers and users.

Summary:
A computer is a machine that can be programmed to carry out sequences of arithmetic or logical operations automatically. Modern digital electronic computers can perform generic sets of operations known as programs. These programs enable computers to perform a wide range of tasks. A computer system is a nominally complete computer that includes the hardware, operating system, and peripheral equipment.

In [ ]:
!pip install rouge
from rouge import Rouge

# Your original text and generated summary
original_text = text
generated_summary = summary_text

# Initialize the Rouge object
rouge = Rouge()

# Calculate ROUGE scores
scores = rouge.get_scores(generated_summary, original_text)

# Print the scores
print("ROUGE Scores:")
print(scores)

ROUGE Scores:
[{'rouge-1': {'r': 0.46875, 'p': 1.0, 'f': 0.6382978679945678}, 'rouge-2': {'r': 0.40425531914893614, 'p': 1.0, 'f': 0.575757571657484}, 'rouge-l': {'r': 0.46875, 'p': 1.0, 'f': 0.6382978679945678}}]


ROUGE-1:

Precision (p): 1.000
Recall (r): 0.469
F1-score (f): 0.638

ROUGE-2:

Precision (p): 1.000
Recall (r): 0.404
F1-score (f): 0.576

ROUGE-L:

Precision (p): 1.000
Recall (r): 0.469
F1-score (f): 0.638

These scores indicate a high level of precision, suggesting that the generated summary contains a substantial amount of correct n-grams. However, the recall values are not as high, indicating that there might be some important content in the original text that is not captured by the generated summary.



In [ ]:
# Load the content of the text file
file_path = '/content/extracted_content/example/space.txt'
with open(file_path, 'r', encoding='utf-8') as file:
    text = file.read()

# Tokenize and encode the input text
inputs = tokenizer.encode("summarize: " + text, return_tensors="pt", max_length=1024, truncation=True)

# Generate the summary using the BART model
summary_ids = model.generate(inputs, max_length=150, length_penalty=2.0, num_beams=4, early_stopping=True)

# Decode the generated summary and print it
summary_text = tokenizer.decode(summary_ids[0], skip_special_tokens=True)
print("Original Text:")
print(text)
print("\nSummary:")
print(summary_text)

Original Text:
A white hole is a bizarre cosmic object which is intensely bright, and from which matter gushes rather than disappears. In other words, it’s the exact opposite of a black hole. But unlike black holes, there’s no consensus about whether white holes exist, or how they’d be formed.They are predicted by Einstein’s theory of gravity, and are most often mentioned in the context of ‘wormholes’, in which a black hole acts as the entry point to a tunnel through space and time, ending in a white hole somewhere else in the Universe. But this is deeply controversial, because Einstein’s theory predicts the existence of a so-called singularity at the centre of black holes – a state of infinite gravity which would prevent anything from passing through to the white hole on the other side. The process starts when an old massive star collapses under its own weight and forms a black hole. But then, quantum effects occurring around the surface of the black hole halt further collapse to a si

Original Text:
A white hole is a bizarre cosmic object which is intensely bright, and from which matter gushes rather than disappears. In other words, it’s the exact opposite of a black hole. But unlike black holes, there’s no consensus about whether white holes exist, or how they’d be formed.They are predicted by Einstein’s theory of gravity, and are most often mentioned in the context of ‘wormholes’, in which a black hole acts as the entry point to a tunnel through space and time, ending in a white hole somewhere else in the Universe. But this is deeply controversial, because Einstein’s theory predicts the existence of a so-called singularity at the centre of black holes – a state of infinite gravity which would prevent anything from passing through to the white hole on the other side. The process starts when an old massive star collapses under its own weight and forms a black hole. But then, quantum effects occurring around the surface of the black hole halt further collapse to a singularity, and instead begin to gradually turn the black hole into a white hole that’s spewing out the original star matter again. The process is mind-bendingly slow, though, so we may be in for a very long wait to find out if white holes really exist.

Summary:
Unlike black holes, there’s no consensus about whether white holes exist, or how they’d be formed. The process starts when an old massive star collapses under its own weight and forms a black hole. But then, quantum effects occurring around the surface of the black hole halt further collapse to a singularity, and instead begin to gradually turn it into a white hole.

In [ ]:
!pip install rouge
from rouge import Rouge

# Your original text and generated summary
original_text = text
generated_summary = summary_text

# Initialize the Rouge object
rouge = Rouge()

# Calculate ROUGE scores
scores = rouge.get_scores(generated_summary, original_text)

# Print the scores
print("ROUGE Scores:")
print(scores)

ROUGE Scores:
[{'rouge-1': {'r': 0.375, 'p': 0.9622641509433962, 'f': 0.5396825356468185}, 'rouge-2': {'r': 0.2893401015228426, 'p': 0.9344262295081968, 'f': 0.44186046150561864}, 'rouge-l': {'r': 0.375, 'p': 0.9622641509433962, 'f': 0.5396825356468185}}]


ROUGE-1:

Precision (p): 0.962
Recall (r): 0.375
F1-score (f): 0.540

ROUGE-2:

Precision (p): 0.934
Recall (r): 0.289
F1-score (f): 0.442

ROUGE-L:

Precision (p): 0.962
Recall (r): 0.375
F1-score (f): 0.540

These scores indicate that the generated summary has high precision but lower recall. In other words, the generated summary includes a significant portion of correct n-grams, but it might miss some important content present in the original text.

## Pipeline Method

In [ ]:
import torch
from transformers import BartForConditionalGeneration, BartTokenizer
from transformers import pipeline

summarizer = pipeline("summarization", model="facebook/bart-large-cnn")

In [ ]:
# Load the content of the text file
file_path = '/content/extracted_content/example/msft.txt'
with open(file_path, 'r', encoding='utf-8') as file:
    text = file.read()

print("Original Text:")
print(text)
print("\nSummary:")
print(summarizer(text, max_length=200, min_length=30, do_sample=False))


Original Text:
In an attempt to build an AI-ready workforce, Microsoft announced Intelligent Cloud Hub which has been launched to empower the next generation of students with AI-ready skills. Envisioned as a three-year collaborative program, Intelligent Cloud Hub will support around 100 institutions with AI infrastructure, course content and curriculum, developer support, development tools and give students access to cloud and AI services. As part of the program, the Redmond giant which wants to expand its reach and is planning to build a strong developer ecosystem in India with the program will set up the core AI infrastructure and IoT Hub for the selected campuses. The company will provide AI development tools and Azure AI services such as Microsoft Cognitive Services, Bot Services and Azure Machine Learning.According to Manish Prakash, Country General Manager-PS, Health and Education, Microsoft India, said, "With AI being the defining technology of our time, it is transforming lives

Original Text:
In an attempt to build an AI-ready workforce, Microsoft announced Intelligent Cloud Hub which has been launched to empower the next generation of students with AI-ready skills. Envisioned as a three-year collaborative program, Intelligent Cloud Hub will support around 100 institutions with AI infrastructure, course content and curriculum, developer support, development tools and give students access to cloud and AI services. As part of the program, the Redmond giant which wants to expand its reach and is planning to build a strong developer ecosystem in India with the program will set up the core AI infrastructure and IoT Hub for the selected campuses. The company will provide AI development tools and Azure AI services such as Microsoft Cognitive Services, Bot Services and Azure Machine Learning.According to Manish Prakash, Country General Manager-PS, Health and Education, Microsoft India, said, "With AI being the defining technology of our time, it is transforming lives and industry and the jobs of tomorrow will require a different skillset. This will require more collaborations and training and working with AI. That’s why it has become more critical than ever for educational institutions to integrate new cloud and AI technologies. The program is an attempt to ramp up the institutional set-up and build capabilities among the educators to educate the workforce of tomorrow." The program aims to build up the cognitive skills and in-depth understanding of developing intelligent cloud connected solutions for applications across industry. Earlier in April this year, the company announced Microsoft Professional Program In AI as a learning track open to the public. The program was developed to provide job ready skills to programmers who wanted to hone their skills in AI and data science with a series of online courses which featured hands-on labs and expert instructors as well. This program also included developer-focused AI school that provided a bunch of assets to help build AI skills.

Summary:
Intelligent Cloud Hub will support around 100 institutions with AI infrastructure, course content and curriculum. Redmond giant wants to expand its reach and is planning to build a strong developer ecosystem in India. The company will provide AI development tools and Azure AI services.

In [ ]:
# Load the content of the text file
file_path = '/content/extracted_content/example/computer.txt'
with open(file_path, 'r', encoding='utf-8') as file:
    text = file.read()

print("Original Text:")
print(text)
print("\nSummary:")
print(summarizer(text, max_length=150, min_length=30, do_sample=False))

Original Text:
A computer is a machine that can be programmed to carry out sequences of arithmetic or logical operations automatically. Modern digital electronic computers can perform generic sets of operations known as programs. These programs enable computers to perform a wide range of tasks. A computer system is a nominally complete computer that includes the hardware, operating system, and peripheral equipment needed and used for full operation. This term may also refer to a group of computers that are linked and function together, such as a computer network or computer cluster. A broad range of industrial and consumer products use computers as control systems. Simple special-purpose devices like microwave ovens and remote controls are included, as are factory devices like industrial robots and computer-aided design, as well as general-purpose devices like personal computers and mobile devices like smartphones. Computers power the Internet, which links billions of other computers a

Original Text:
A computer is a machine that can be programmed to carry out sequences of arithmetic or logical operations automatically. Modern digital electronic computers can perform generic sets of operations known as programs. These programs enable computers to perform a wide range of tasks. A computer system is a nominally complete computer that includes the hardware, operating system, and peripheral equipment needed and used for full operation. This term may also refer to a group of computers that are linked and function together, such as a computer network or computer cluster. A broad range of industrial and consumer products use computers as control systems. Simple special-purpose devices like microwave ovens and remote controls are included, as are factory devices like industrial robots and computer-aided design, as well as general-purpose devices like personal computers and mobile devices like smartphones. Computers power the Internet, which links billions of other computers and users.

Summary:
A computer is a machine that can be programmed to carry out sequences of arithmetic or logical operations automatically. Modern digital electronic computers can perform generic sets of operations known as programs. A computer system is a nominally complete computer that includes the hardware, operating system, and peripheral equipment needed and used for full operation.

In [ ]:
# Load the content of the text file
file_path = '/content/extracted_content/example/space.txt'
with open(file_path, 'r', encoding='utf-8') as file:
    text = file.read()

print("Original Text:")
print(text)
print("\nSummary:")
print(summarizer(text, max_length=130, min_length=30, do_sample=False))


Original Text:
A white hole is a bizarre cosmic object which is intensely bright, and from which matter gushes rather than disappears. In other words, it’s the exact opposite of a black hole. But unlike black holes, there’s no consensus about whether white holes exist, or how they’d be formed.They are predicted by Einstein’s theory of gravity, and are most often mentioned in the context of ‘wormholes’, in which a black hole acts as the entry point to a tunnel through space and time, ending in a white hole somewhere else in the Universe. But this is deeply controversial, because Einstein’s theory predicts the existence of a so-called singularity at the centre of black holes – a state of infinite gravity which would prevent anything from passing through to the white hole on the other side. The process starts when an old massive star collapses under its own weight and forms a black hole. But then, quantum effects occurring around the surface of the black hole halt further collapse to a si

Original Text:
A white hole is a bizarre cosmic object which is intensely bright, and from which matter gushes rather than disappears. In other words, it’s the exact opposite of a black hole. But unlike black holes, there’s no consensus about whether white holes exist, or how they’d be formed.They are predicted by Einstein’s theory of gravity, and are most often mentioned in the context of ‘wormholes’, in which a black hole acts as the entry point to a tunnel through space and time, ending in a white hole somewhere else in the Universe. But this is deeply controversial, because Einstein’s theory predicts the existence of a so-called singularity at the centre of black holes – a state of infinite gravity which would prevent anything from passing through to the white hole on the other side. The process starts when an old massive star collapses under its own weight and forms a black hole. But then, quantum effects occurring around the surface of the black hole halt further collapse to a singularity, and instead begin to gradually turn the black hole into a white hole that’s spewing out the original star matter again. The process is mind-bendingly slow, though, so we may be in for a very long wait to find out if white holes really exist.

Summary:
A white hole is a cosmic object which is intensely bright, and from which matter gushes rather than disappears. They are predicted by Einstein’s theory of gravity, and are most often mentioned in the context of ‘wormholes’, in which a black hole acts as the entry point to a tunnel through space and time, ending in a white hole somewhere else in the Universe. But unlike black holes, there's no consensus about whether white holes exist, or how they’d be formed.

In [ ]:
text = """ New York (CNN)When Liana Barrientos was 23 years old, she got married in Westchester County, New York.
A year later, she got married again in Westchester County, but to a different man and without divorcing her first husband.
Only 18 days after that marriage, she got hitched yet again. Then, Barrientos declared "I do" five more times, sometimes only within two weeks of each other.
In 2010, she married once more, this time in the Bronx. In an application for a marriage license, she stated it was her "first and only" marriage.
Barrientos, now 39, is facing two criminal counts of "offering a false instrument for filing in the first degree," referring to her false statements on the
2010 marriage license application, according to court documents.
Prosecutors said the marriages were part of an immigration scam.
On Friday, she pleaded not guilty at State Supreme Court in the Bronx, according to her attorney, Christopher Wright, who declined to comment further.
After leaving court, Barrientos was arrested and charged with theft of service and criminal trespass for allegedly sneaking into the New York subway through an emergency exit, said Detective
Annette Markowski, a police spokeswoman. In total, Barrientos has been married 10 times, with nine of her marriages occurring between 1999 and 2002.
All occurred either in Westchester County, Long Island, New Jersey or the Bronx. She is believed to still be married to four men, and at one time, she was married to eight men at once, prosecutors say.
Prosecutors said the immigration scam involved some of her husbands, who filed for permanent residence status shortly after the marriages.
Any divorces happened only after such filings were approved. It was unclear whether any of the men will be prosecuted.
The case was referred to the Bronx District Attorney\'s Office by Immigration and Customs Enforcement and the Department of Homeland Security\'s
Investigation Division. Seven of the men are from so-called "red-flagged" countries, including Egypt, Turkey, Georgia, Pakistan and Mali.
Her eighth husband, Rashid Rajput, was deported in 2006 to his native Pakistan after an investigation by the Joint Terrorism Task Force.
If convicted, Barrientos faces up to four years in prison.  Her next court appearance is scheduled for May 18.
"""
print("Original Text:")
print(text)
print("\nSummary:")
print(summarizer(text, max_length=150, min_length=30, do_sample=False))


Original Text:
 New York (CNN)When Liana Barrientos was 23 years old, she got married in Westchester County, New York.
A year later, she got married again in Westchester County, but to a different man and without divorcing her first husband.
Only 18 days after that marriage, she got hitched yet again. Then, Barrientos declared "I do" five more times, sometimes only within two weeks of each other.
In 2010, she married once more, this time in the Bronx. In an application for a marriage license, she stated it was her "first and only" marriage.
Barrientos, now 39, is facing two criminal counts of "offering a false instrument for filing in the first degree," referring to her false statements on the
2010 marriage license application, according to court documents.
Prosecutors said the marriages were part of an immigration scam.
On Friday, she pleaded not guilty at State Supreme Court in the Bronx, according to her attorney, Christopher Wright, who declined to comment further.
After leaving co

Original Text:
 New York (CNN)When Liana Barrientos was 23 years old, she got married in Westchester County, New York.
A year later, she got married again in Westchester County, but to a different man and without divorcing her first husband.
Only 18 days after that marriage, she got hitched yet again. Then, Barrientos declared "I do" five more times, sometimes only within two weeks of each other.
In 2010, she married once more, this time in the Bronx. In an application for a marriage license, she stated it was her "first and only" marriage.
Barrientos, now 39, is facing two criminal counts of "offering a false instrument for filing in the first degree," referring to her false statements on the
2010 marriage license application, according to court documents.
Prosecutors said the marriages were part of an immigration scam.
On Friday, she pleaded not guilty at State Supreme Court in the Bronx, according to her attorney, Christopher Wright, who declined to comment further.
After leaving court, Barrientos was arrested and charged with theft of service and criminal trespass for allegedly sneaking into the New York subway through an emergency exit, said Detective
Annette Markowski, a police spokeswoman. In total, Barrientos has been married 10 times, with nine of her marriages occurring between 1999 and 2002.
All occurred either in Westchester County, Long Island, New Jersey or the Bronx. She is believed to still be married to four men, and at one time, she was married to eight men at once, prosecutors say.
Prosecutors said the immigration scam involved some of her husbands, who filed for permanent residence status shortly after the marriages.
Any divorces happened only after such filings were approved. It was unclear whether any of the men will be prosecuted.
The case was referred to the Bronx District Attorney's Office by Immigration and Customs Enforcement and the Department of Homeland Security's
Investigation Division. Seven of the men are from so-called "red-flagged" countries, including Egypt, Turkey, Georgia, Pakistan and Mali.
Her eighth husband, Rashid Rajput, was deported in 2006 to his native Pakistan after an investigation by the Joint Terrorism Task Force.
If convicted, Barrientos faces up to four years in prison.  Her next court appearance is scheduled for May 18.

Summary : Liana Barrientos, 39, is charged with two counts of "offering a false instrument for filing in the first degree" In total, she has been married 10 times, with nine of her marriages occurring between 1999 and 2002. She is believed to still be married to four men.

In [ ]:
text = """ A Namibian cheetah has given birth to three cubs at the Kuno National Park in Madhya Pradesh, weeks after three cubs were born to another feline.

Union Environment Minister Bhupender Yadav shared the news on X, saying, "Kuno's new cubs! Namibian Cheetah named Jwala has given birth to three cubs. This comes just weeks after Namibian Cheetah Aasha gave birth to her cubs." He further said in the post, "Congratulations to all wildlife frontline warriors and wildlife lovers across the country. May Bharat's wildlife thrive." On January 3, Kuno National Park officials reported that Namibian cheetah Aasha had given birth to three cubs.

Before this, Jwala (Namibian name Siyaya) had given birth to four cubs last March. However, only one of them survived.

Jwala and Aasha are among the cheetahs translocated to India from Namibia under Prime Minister Narendra Modi's 'Project Cheetah', aimed at re-introducing the only large carnivore species that went extinct in independent India.

The first batch of eight cheetahs was introduced in India in September 2022.

The second batch of 12 cheetahs was flown in from South Africa last February."""

print("Original Text:")
print(text)
print("\nSummary:")
print(summarizer(text, max_length=150, min_length=30, do_sample=False))

Original Text:
 A Namibian cheetah has given birth to three cubs at the Kuno National Park in Madhya Pradesh, weeks after three cubs were born to another feline.

Union Environment Minister Bhupender Yadav shared the news on X, saying, "Kuno's new cubs! Namibian Cheetah named Jwala has given birth to three cubs. This comes just weeks after Namibian Cheetah Aasha gave birth to her cubs." He further said in the post, "Congratulations to all wildlife frontline warriors and wildlife lovers across the country. May Bharat's wildlife thrive." On January 3, Kuno National Park officials reported that Namibian cheetah Aasha had given birth to three cubs.

Before this, Jwala (Namibian name Siyaya) had given birth to four cubs last March. However, only one of them survived.

Jwala and Aasha are among the cheetahs translocated to India from Namibia under Prime Minister Narendra Modi's 'Project Cheetah', aimed at re-introducing the only large carnivore species that went extinct in independent India.

Original Text:
 A Namibian cheetah has given birth to three cubs at the Kuno National Park in Madhya Pradesh, weeks after three cubs were born to another feline.

Union Environment Minister Bhupender Yadav shared the news on X, saying, "Kuno's new cubs! Namibian Cheetah named Jwala has given birth to three cubs. This comes just weeks after Namibian Cheetah Aasha gave birth to her cubs." He further said in the post, "Congratulations to all wildlife frontline warriors and wildlife lovers across the country. May Bharat's wildlife thrive." On January 3, Kuno National Park officials reported that Namibian cheetah Aasha had given birth to three cubs.

Before this, Jwala (Namibian name Siyaya) had given birth to four cubs last March. However, only one of them survived.

Jwala and Aasha are among the cheetahs translocated to India from Namibia under Prime Minister Narendra Modi's 'Project Cheetah', aimed at re-introducing the only large carnivore species that went extinct in independent India.

The first batch of eight cheetahs was introduced in India in September 2022.

The second batch of 12 cheetahs was flown in from South Africa last February.

Summary:
A Namibian cheetah has given birth to three cubs at the Kuno National Park in Madhya Pradesh. On January 3, Kuno national Park officials reported that Namibians Aasha and Jwala had given birth. The first batch of eight cheetahs was introduced in India in September 2022.

In [ ]:
text = """Firm GDP growth forecast, inflation at manageable levels, political stability at the central government level, and signs that the central banks the world over are done with their monetary policy tightening have painted a bright picture for India - which many agencies have termed to be the fastest-growing major economy.

The strong inflow of funds from foreign portfolio investors (FPIs) lately also supported the stocks to march towards all-time highs. Notably, foreign portfolio investors have again trained their sight towards India, becoming net buyers in the country's stock market. In the process, it helped Indian benchmark stock indices taste their all-time highs recently.

India, which last year became the most populous country has positioned itself as an alternative to China, attracting fresh capital from global investors and companies alike, thanks to its stable political setup and a consumption-driven economy that remains among the fastest-growing of major nations, the Bloomberg report said.

As Indian stocks rallied, it coincided with a historic slump in Hong Kong, where some of China's most influential and innovative firms are listed.

According to the news report, stringent anti-COVID-19 curbs, regulatory crackdowns on corporations, a property-sector crisis and geopolitical tensions with the West have all combined to erode China's appeal as the world's growth engine.

According to the news report, new listings have dried up in Hong Kong, with the Asian financial hub losing its status as one of the world's busiest venues for initial public offerings (IPOs). """

print("Original Text:")
print(text)
print("\nSummary:")
print(summarizer(text, max_length=250, min_length=30, do_sample=False))


Original Text:
Firm GDP growth forecast, inflation at manageable levels, political stability at the central government level, and signs that the central banks the world over are done with their monetary policy tightening have painted a bright picture for India - which many agencies have termed to be the fastest-growing major economy.

The strong inflow of funds from foreign portfolio investors (FPIs) lately also supported the stocks to march towards all-time highs. Notably, foreign portfolio investors have again trained their sight towards India, becoming net buyers in the country's stock market. In the process, it helped Indian benchmark stock indices taste their all-time highs recently.

India, which last year became the most populous country has positioned itself as an alternative to China, attracting fresh capital from global investors and companies alike, thanks to its stable political setup and a consumption-driven economy that remains among the fastest-growing of major nations, 

Original Text:
 Firm GDP growth forecast, inflation at manageable levels, political stability at the central government level, and signs that the central banks the world over are done with their monetary policy tightening have painted a bright picture for India - which many agencies have termed to be the fastest-growing major economy.

The strong inflow of funds from foreign portfolio investors (FPIs) lately also supported the stocks to march towards all-time highs. Notably, foreign portfolio investors have again trained their sight towards India, becoming net buyers in the country's stock market. In the process, it helped Indian benchmark stock indices taste their all-time highs recently.

India, which last year became the most populous country has positioned itself as an alternative to China, attracting fresh capital from global investors and companies alike, thanks to its stable political setup and a consumption-driven economy that remains among the fastest-growing of major nations, the Bloomberg report said.

As Indian stocks rallied, it coincided with a historic slump in Hong Kong, where some of China's most influential and innovative firms are listed.

According to the news report, stringent anti-COVID-19 curbs, regulatory crackdowns on corporations, a property-sector crisis and geopolitical tensions with the West have all combined to erode China's appeal as the world's growth engine.

According to the news report, new listings have dried up in Hong Kong, with the Asian financial hub losing its status as one of the world's busiest venues for initial public offerings (IPOs).

Summary:
The world's fastest-growing economy, India, has seen a surge in its stock market. The country's economy is expected to grow at a faster rate than the rest of the world.

## REFERENCES
https://blog.paperspace.com/bart-model-for-text-summarization-part1/

https://keras.io/examples/nlp/abstractive_summarization_with_bart/
